pkgs/development/python-modules/yapf/default.nix

```diff
@@ -1,19 +1,56 @@
-{ lib, buildPythonPackage, fetchPypi }:
+{ lib
+, buildPythonApplication
+, fetchFromGitHub
+, nose
+, setuptools
+}:
 
-buildPythonPackage rec {
+buildPythonApplication rec {
   pname = "yapf";
-  version = "0.30.0";
+  version = "0.31.0";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "3000abee4c28daebad55da6c85f3cd07b8062ce48e2e9943c8da1b9667d48427";
+  src = fetchFromGitHub {
+    owner = "google";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-Yfc1ES16MT6IpqzBD79F1U/d+32O7KKWYm7nfS8LqcQ=";
   };
 
+  nativeBuildInputs = [
+    setuptools
+  ];
+  checkInputs = [
+    nose
+  ];
+
+  checkPhase = ''
+    nosetests
+  '';
+
   meta = with lib; {
-    description = "A formatter for Python code.";
-    homepage    = "https://github.com/google/yapf";
-    license     = licenses.asl20;
-    maintainers = with maintainers; [ siddharthist ];
-  };
+    homepage = "https://github.com/google/yapf";
+    description = "Yet Another Python Formatter";
+    longDescription = ''
+      Most of the current formatters for Python --- e.g., autopep8, and pep8ify
+      --- are made to remove lint errors from code. This has some obvious
+      limitations. For instance, code that conforms to the PEP 8 guidelines may
+      not be reformatted. But it doesn't mean that the code looks good.
 
+      YAPF takes a different approach. It's based off of 'clang-format',
+      developed by Daniel Jasper. In essence, the algorithm takes the code and
+      reformats it to the best formatting that conforms to the style guide, even
+      if the original code didn't violate the style guide. The idea is also
+      similar to the 'gofmt' tool for the Go programming language: end all holy
+      wars about formatting - if the whole codebase of a project is simply piped
+      through YAPF whenever modifications are made, the style remains consistent
+      throughout the project and there's no point arguing about style in every
+      code review.
+
+      The ultimate goal is that the code YAPF produces is as good as the code
+      that a programmer would write if they were following the style guide. It
+      takes away some of the drudgery of maintaining your code.
+    '';
+    license = licenses.asl20;
```

Comment:
```suggestion
    license     = licenses.asl20;
```

Reply:
Pretty-formatting like this is superfluous.

Reply:
It is. You should just keep the diff as small as possible.